components/usage/pkg/apiv1/billing.go

```diff
@@ -165,6 +167,80 @@ func (s *BillingService) CreateStripeCustomer(ctx context.Context, req *v1.Creat
 	}, nil
 }
 
+func (s *BillingService) CreateStripeSubscription(ctx context.Context, req *v1.CreateStripeSubscriptionRequest) (*v1.CreateStripeSubscriptionResponse, error) {
+	attributionID, err := db.ParseAttributionID(req.GetAttributionId())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid attribution ID %s", attributionID)
+	}
+
+	customer, err := s.GetStripeCustomer(ctx, &v1.GetStripeCustomerRequest{Identifier: &v1.GetStripeCustomerRequest_AttributionId{AttributionId: string(attributionID)}})
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Stripe customer with attribution ID %s not found", attributionID)
+	}
+
+	// if req.CustomerId == "" {
+	// 	return nil, status.Error(codes.InvalidArgument, "Invalid customer ID")
+	// }
+
+	_, err = s.stripeClient.SetDefaultPaymentForCustomer(ctx, customer.Customer.Id, req.SetupIntentId)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Failed to set default payment for customer ID %s", customer.Customer.Id)
+	}
+
+	stripeCustomer, err := s.stripeClient.GetCustomer(ctx, customer.Customer.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	priceIdentifier := getPriceIdentifier(attributionID, stripeCustomer, s)
+	if priceIdentifier == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid currency %s for customer ID %s", stripeCustomer.Metadata["preferredCurrency"], stripeCustomer.ID)
+	}
+
+	var isAutomaticTaxSupported bool
+	if stripeCustomer.Tax != nil {
+		isAutomaticTaxSupported = stripeCustomer.Tax.AutomaticTax == "supported"
+	}
+	if !isAutomaticTaxSupported {
+		log.Warnf("Automatic Stripe tax is not supported for customer %s", stripeCustomer.ID)
+	}
+
+	subscription, err := s.stripeClient.CreateSubscription(ctx, stripeCustomer.ID, priceIdentifier, isAutomaticTaxSupported)
+	if err != nil {
+		return nil, status.Errorf(codes.Aborted, "Failed to create subscription with customer ID %s", customer.Customer.Id)
+	}
+
+	return &v1.CreateStripeSubscriptionResponse{
+		Subscription: &v1.StripeSubscription{
+			Id: subscription.ID,
+		},
+	}, nil
+}
+
+func getPriceIdentifier(attributionID db.AttributionID, stripeCustomer *stripe_api.Customer, s *BillingService) string {
+	priceIdentifier := ""
+
+	if stripeCustomer.Metadata["preferredCurrency"] == "" {
+		log.Errorf(">>>>> Currency is not available")
```

Comment:
Debug log?

Reply:
Will remove the ">>>>>" but I intend to keep this log. Would that be ok?

Reply:
`log.WithField("stripe_customer_id", customer.ID).WithField("stripe_preferred_currency", stripeCustomer.Metadata["preferredCurrency"]).Warn("No preferred currency set. Defaulting to X")`